Orange/widgets/data/owcorrelations.py

```diff
@@ -426,8 +426,12 @@ def send_report(self):
     def migrate_context(cls, context, version):
         if version < 2:
             sel = context.values["selection"]
-            context.values["selection"] = ([(var.name, vartype(var))
-                                            for var in sel[0]], sel[1])
+            context.values["selection"] = [(var.name, vartype(var))
+                                           for var in sel[0]]
+        if version < 3:
+            sel = context.values["selection"]
+            context.values["selection"] = ([(name, vtype + 100)
+                                            for name, vtype in sel], -3)
```

Comment:
I am confused, isn't selection now supposed to be a list of variables?

Reply:
The widget's attribute (as in `widget.selection`) is a list of `Variables`. Context handler encodes it (as in `context.values["selection"]`) as a list of (name, type) pairs.

In settings version 2 the widget had an attribute `selection` which contains a list of (name, type) pairs. This was safe for saving, but does the encoding within the widget instead of in the handler. The handler thus saw strings with names of variables and issued a warning.

In settings version 3, the widget's `selection` is a list of `Variables`, and encoding is done within the handler.

Coincidentally, the encoded setting looks (almost) the same, because the widget's encoding was (almost) the same as encoding by the handler. So migration just adds `100` to vartypes and appends `-3` to the tuple to mark it as a list.